### PR TITLE
v4.0.1-alpha.2

### PR DESCRIPTION
**Pull request by Spacesai1or**
#### web3-eth-contract

-   Decoding error data, using Error ABI if available, if error was returned from a smart contract function call (#5662).

#### web3-types

-   These types were moved from `web3-eth-accounts` to `web3-types` package: Cipher, CipherOptions, ScryptParams, PBKDF2SHA256Params, KeyStore (#5581 )

#### web3-utils

-   Export a new function `uuidV4` that generates a random v4 Uuid (#5373).
-   Enable passing a starting number, to increment based on it, for the Json Rpc Request `id` (#5652).
-   Export a new function `isPromise` that checks if an object is a promise (#5652).

#### web3-eth-contract

-   `SpecialOutput` type was added as a generic type into the call function to support reassigning output types (#5631)
-   Overloaded functions types (`ContractOverloadedMethodInputs`, `ContractOverloadedMethodOutputs`) was added (#5631)

#### web3-eth-contract

-   Emit past contract events based on `fromBlock` when passed to `contract.events.someEventName` (#5201)
-   Use different types for `ContractOptions` -> `jsonInterface` setter and getter (#5474)
-   An issue within the `Contract` constructor where `provider` wasn't being set when provided within the `optionsOrContextOrReturnFormat` argument (#5669)

#### web3-types

-   Make the `request` method of `EIP1193Provider` class, compatible with EIP 1193 (#5591)

#### web3-utils

-   Use Uuid for the response id, to fix the issue "Responses get mixed up due to conflicting payload IDs" (#5373).

#### web3-validator

-   Fix `isHex`returning `false` for `-123`, fix `isHexStrict` returning `true` for `-0x`, and fix `isHex` returning `true` for empty strings `` (#5373).

#### web3-eth-abi

-   Fix ContractMethodOutputParameters type to support output object types by index and string key. Also, it returns void if ABI doesn't have outputs and returns exactly one type if the output array has only one element. (#5631)

#### web3-eth-accounts

-   These types were moved from `web3-eth-accounts` to `web3-types` package: Cipher, CipherOptions, ScryptParams, PBKDF2SHA256Params, KeyStore (#5581 )

### CODE REVIEW

File: tools/web3-packagetemplate/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "web3-packagetemplate",
-	"version": "0.1.1-alpha.0",
+	"version": "0.1.1-alpha.1",
```
Comment (jdevcs) [inline tools/web3-packagetemplate/package.json]: I think there isn't any change in this package ?, so we dnt need version bump.  Its easier to review if we include changelog changes in same PR.

**Review by avkos — APPROVED** [review]

**Comment by avkos** [comment]
@spacesailor24 thanks for PR. just need to merge 4.x into this branch and resolve conflicts.

**Comment by Spacesai1or** [comment]
> @spacesailor24 thanks for PR. just need to merge 4.x into this branch and resolve conflicts.

@avkos So usually we would have an RC version we could bump every time we make changes to the release since the last commit in the release is tagged with the version. We've already messed up by merging #5689 into this branch, since now the branch has commits that come after the tagged commit. This isn't a huge issue since this only really affects the continuity of the [release listed in Github](https://github.com/web3/web3.js/releases/tag/untagged-e2800057933dc0d92b7d) (I'm talking about the `Assets` listed in the release). I believe the right steps would've been to open a PR that merges `4.x` into the release branch to solve the conflicts and cut another RC, but we don't have that option now since there are no RC versions for `4.x` yet

So, if we are to merge `4.x` into the release branch and solve the conflicts, we'd need to re-build all the packages, commit, tag the commit, and re-create the release on Github

**Comment by jdevcs** [comment]
> > @spacesailor24 thanks for PR. just need to merge 4.x into this branch and resolve conflicts.
> 
> @avkos So usually we would have an RC version we could bump every time we make changes to the release since the last commit in the release is tagged with the version. We've already messed up by merging #5689 into this branch, since now the branch has commits that come after the tagged commit. This isn't a huge issue since this only really affects the continuity of the [release listed in Github](https://github.com/web3/web3.js/releases/tag/untagged-e2800057933dc0d92b7d) (I'm talking about the `Assets` listed in the release). I believe the right steps would've been to open a PR that merges `4.x` into the release branch to solve the conflicts and cut another RC, but we don't have that option now since there are no RC versions for `4.x` yet
> 
> So, if we are to merge `4.x` into the release branch and solve the conflicts, we'd need to re-build all the packages, commit, tag the commit, and re-create the release on Github

In case of merge conflicts, and before actual release, could you go ahead with following:
Merge main branch into release PR, build, delete old tag, recreate new tag, and re populate github draft release with newly created tag. There isn't any technical limitation or issue in that, if it happens before release on npm / github. 

In future we should also follow freezing main branch for 4.x if there is release PR created, we were following this in 1.x.

File: packages/web3-eth-abi/CHANGELOG.md [hunk 2]
```diff
@@ -46,6 +46,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 -   Return `BigInt` instead of `string` when decoding function parameters for large numbers, such as `uint256`. (#5435)
 
+## [4.0.1-alpha.2]
+
+### Changed
+
+-   Updated Web3.js dependencies (#5664)
+
 ## [Unreleased]
```
Comment (jdevcs) [inline packages/web3-eth-abi/CHANGELOG.md]: This should be at end of file

File: packages/web3-validator/CHANGELOG.md [hunk 3]
```diff
@@ -41,6 +41,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 -   Removed direct function `toJSON()` in `Web3ValidatorError` class as its available via base class (#5435)
 
+## [0.1.1-alpha.2]
+
+### Changed
+
+-   Updated Web3.js dependencies (#5664)
+
 ## [Unreleased]
```
Comment (jdevcs) [inline packages/web3-validator/CHANGELOG.md]: This should be at end of file

**Review by jdevcs — APPROVED** [review]

**Review by luu-alex — APPROVED** [review]

**Pull request merged**